test/unit/NodeCompiler/CompileNodeToValueTest.php

```diff
@@ -23,6 +23,10 @@ public function nodeProvider()
             ['[]', []],
             ['false', false],
             ['true', true],
+            ['[1,2,3]', [1, 2, 3]],
+            ['["foo","bar"]', ['foo', 'bar']],
+            ['[1 => "foo", 2 => "bar"]', [1 => 'foo', 2 => 'bar']],
+            ['["foo" => "bar"]', ['foo' => 'bar']],
```

Comment:
There's no point, there could be loads of tests we could add here. Issue will suffice for this, otherwise that test will be one massive block of `@todo` comments